### DISCUSSION TITLE
USB transaction failed warnings with AKAI MPK Mini MIDI controller

### DISCUSSION THREAD

**Discussion by daparker26** [rsta2/circle#340] [Q&A] [answered]
Hi Rene and anyone else who could help!

I am using Circle Step 45 on a Raspberry Pi 1 (B2). I'm trying to use an [AKAI MPK Mini 3](https://www.akaipro.com/mpk-mini-mk3). However, once Circle recognizes it, I get a string of these warnings:

`00:00:06.44 dwhci: Transaction failed (status 0x202)`. 
`00:00:06.44 dwhci: Transaction failed (status 0x202)`. 
`00:00:06.44 dwhci: Transaction failed (status 0x202)`. 
`00:00:06.45 dwhci: Transaction failed (status 0x202)`. 

It seems to still work—it detects when I press the keys, but these messages are concerning. I assume I can just suppress them, but I wondered if there was a fix or workaround? I don't know a lot about USB, so I don't really know where to begin. I've added `usbspeed=full` in my cmdline.txt file, but it doesn't help. Thanks!

**Comment by rsta2** [comment, marked as answer]
Hi Daniel, this problem is caused by an unfavorable timing on the USB. These messages are not critical, but it is recommended to enable the system option `REALTIME` or even better `USE_USB_FIQ` by adding the following line to the file *Config.mk* in your project root:

```
DEFINE += -DREALTIME

or

DEFINE += -DUSE_USB_FIQ
```

Everything must be rebuild, after adding this.

`REALTIME` suppresses log messages from IRQ handlers. Because IRQ handlers cannot be interrupted by another IRQ, the system timing may be bad, when a log message is generated in an IRQ handler and for example, the screen has to be scrolled because of this. This happens all inside the IRQ handler and takes milliseconds. With `REALTIME` enabled, this does not happen.

`USE_USB_FIQ` uses the FIQ for the USB host controller driver, which gives the best USB timing and is recommended, but cannot be used, if the FIQ is used for some other purpose (e.g. the serial device).

**Reply by daparker26** [reply]
Thanks, Rene! Enabling REALTIME fixed the issue. However, with REALTIME enabled, the touchscreen is unresponsive. Unfortunately, the touchscreen stopped powering on before I could get around to troubleshooting the issue (I had been having intermittent issues with it since I first got it).
